x/mongo/driver/topology/topology.go

```diff
@@ -363,6 +369,12 @@ func (t *Topology) SelectServerLegacy(ctx context.Context, ss description.Server
 	if atomic.LoadInt32(&t.connectionstate) != connected {
 		return nil, ErrTopologyClosed
 	}
+
```

Comment:
Let's take out these changes. `git grep` shows this function is only used in some low-level integration tests and it shouldn't really exist at this point. I've filed https://jira.mongodb.org/browse/GODRIVER-1702 to remove it.

Reply:
done